src/metabase/query_processor/pivot.clj

```diff
@@ -1,98 +1,144 @@
 (ns metabase.query-processor.pivot
   "Pivot table actions for the query processor"
   (:require [clojure.core.async :as a]
-            [clojure.set :refer [map-invert]]
+            [metabase.mbql.normalize :as normalize]
             [metabase.query-processor :as qp]
             [metabase.query-processor.context :as qp.context]
-            [metabase.query-processor.store :as qp.store]))
+            [metabase.query-processor.context.default :as context.default]
+            [metabase.query-processor.error-type :as qp.error-type]
+            [metabase.query-processor.store :as qp.store]
+            [metabase.util.i18n :refer [tru]]
+            [schema.core :as s]))
 
-(defn powerset
-  "Generate a powerset while maintaining the original ordering as much as possible"
-  [xs]
-  (for [combo (reverse (range (int (Math/pow 2 (count xs)))))]
-    (for [item  (range 0 (count xs))
-          :when (not (zero? (bit-and (bit-shift-left 1 item) combo)))]
-      (nth xs item))))
+(defn- group-number
+  "Come up with a display name given a combination of breakout `indecies` e.g.
 
-(defn- generate-specified-breakouts
-  "Generate breakouts specified by pivot_rows"
-  [breakouts pivot-rows pivot-cols]
-  (let [generator (fn [coll]
-                    (for [pivots coll]
-                      (map #(nth breakouts %) pivots)))]
-    (concat
-     [breakouts]
-     (generator [pivot-cols])
-     (when (and (seq pivot-cols)
-                (seq pivot-rows))
-       (generator [(concat [(first pivot-rows)] pivot-cols)]))
-     (generator (powerset (or pivot-rows
-                              ;; this can happen for the public/embed endpoints,
-                              ;; where we aren't given a pivot_rows / pivot_cols
-                              ;; parameter, so we'll just generate everything
-                              (vec (range 0 (count breakouts)))))))))
+    (group-number 3 [1])   ; -> 5
+    (group-number 3 [1 2]) ; -> 1"
+  [num-breakouts indecies]
+  (transduce
+   (map (partial bit-shift-left 1))
+   (completing bit-xor)
+   (int (dec (Math/pow 2 num-breakouts)))
+   indecies))
```

Comment:
indices

Reply:
wow I really can't spell

Reply:
I can only spell because the red squigglies underneath words nowadays 😁